### PR TITLE
fix: add insertTextFormat to completionItem

### PR DESCRIPTION
**Pull request by yaegassy**
## Description

There is an environment where indentation does not work correctly in `completionItem/resolve`.

It seems to behave correctly in VSCode, but when I tried it in a Vim/Neovim environment, I found a problem. Perhaps the situation is also a problem with other editors and IDEs.

In the case of "TextEdit", the fix was to set the "snippet" format with `insertTextFormat`.

## Before

**[NG] DEMO (mp4)**:

https://user-images.githubusercontent.com/188642/161216267-7c243248-a1e3-44b2-b6fc-ac0bc0bc6da1.mp4

## After

**[OK] DEMO (mp4)**:

https://user-images.githubusercontent.com/188642/161245925-6a16c84c-f2f2-4796-b4cc-4925ad95ed80.mp4

### CODE REVIEW

**Comment by yaegassy** [comment]
@priyamsahoo Could you please check this PR in VSCode?

**Comment by priyamsahoo** [comment]
Thanks for this one, @yaegassy!
This has no side-effects on the vscode extension as well.

The only thing that is missing is, you need to add a change note for this PR. Follow this: https://github.com/ansible/ansible-language-server/blob/5725b7e/docs/changelog-fragments.d/README.md

**Review by priyamsahoo — APPROVED** [review]

**Comment by yaegassy** [comment]
> The only thing that is missing is, you need to add a change note for this PR. Follow this: h ttps://github.com/ansible/ansible-language-server/blob/5725b7e/docs/changelog-fragments.d/README.md

I see. I just don't know what to do...

So can we close my PR and have @priyamsahoo  create a PR for this modification?

**Comment by priyamsahoo** [comment]
> > The only thing that is missing is, you need to add a change note for this PR. Follow this: h ttps://github.com/ansible/ansible-language-server/blob/5725b7e/docs/changelog-fragments.d/README.md
> 
> I see. I just don't know what to do...
> 
> So can we close my PR and have @priyamsahoo create a PR for this modification?

I believe that's not a great idea. Since, you dicovered the bug, I want your PR to be merged so that people can know that you wrote a patch for it. I can help you with writing the change note.

Just go to `docs/changelog-fragments.d/` and create a file named 285.bugfix.md (285 is your PR number in this case).

In that file, add this: 

```
Fixed indentation issue which resolving auto-completion items to support editors like vim and neovim -- by {user}`yaegassy`
```

You can modify the message according to you.

And that's it!

**Comment by yaegassy** [comment]
@priyamsahoo Thanks for helping! Change note has been added.

File: docs/changelog-fragments.d/285.bugfix.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fixed indentation issue which resolving auto-completion items to support editors
```
Comment (priyamsahoo) [inline docs/changelog-fragments.d/285.bugfix.md:1]: replace `which` -> `while`

**Comment by priyamsahoo** [comment]
Just a minor comment.

**Comment by priyamsahoo** [comment]
Also, one more request, could you please paste the log of completionItem after this change, just as a reference ?

**Comment by priyamsahoo** [comment]
Thanks for the help !

**Pull request merged**